app/src/main/java/de/danoeh/antennapod/adapter/SubscriptionsRecyclerAdapter.java

```diff
@@ -0,0 +1,237 @@
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.CheckBox;
+import android.widget.CompoundButton;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.core.text.TextUtilsCompat;
+import androidx.core.view.ViewCompat;
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.lang.ref.WeakReference;
+import java.text.NumberFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.activity.MainActivity;
+import de.danoeh.antennapod.core.feed.LocalFeedUpdater;
+import de.danoeh.antennapod.core.storage.NavDrawerData;
+import de.danoeh.antennapod.fragment.FeedItemlistFragment;
+import de.danoeh.antennapod.fragment.SubscriptionFragment;
+import de.danoeh.antennapod.model.feed.Feed;
+import de.danoeh.antennapod.model.feed.FeedItem;
+import jp.shts.android.library.TriangleLabelView;
+
+/**
+ * Adapter for subscriptions
+ */
+public class SubscriptionsRecyclerAdapter extends SelectableAdapter<SubscriptionsRecyclerAdapter.SubscriptionViewHolder> implements View.OnCreateContextMenuListener
+{
+    /** the position in the view that holds the add item; 0 is the first, -1 is the last position */
+    private static final String TAG = "SubscriptionsRecyclerAdapter";
+
+    private final WeakReference<MainActivity> mainActivityRef;
+    private final ItemAccess itemAccess;
+    private Feed selectedFeed = null;
+    int longPressedPosition = 0; // used to init actionMode
+
+    public SubscriptionsRecyclerAdapter(MainActivity mainActivity, ItemAccess itemAccess) {
+        super(mainActivity);
+        this.mainActivityRef = new WeakReference<>(mainActivity);
+        this.itemAccess = itemAccess;
+    }
+
+    public Object getItem(int position) {
+        return itemAccess.getItem(position);
+    }
+
+    public Feed getSelectedFeed() {
+        return selectedFeed;
+    }
+
+    @NonNull
+    @Override
+    public SubscriptionViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View itemView = LayoutInflater.from(mainActivityRef.get()).inflate(R.layout.subscription_item, parent, false);
+        SubscriptionViewHolder viewHolder = new SubscriptionViewHolder(itemView);
+
+        return viewHolder;
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull SubscriptionViewHolder holder, int position) {
+        holder.onBind(itemAccess.getItem(position), position);
+        holder.itemView.setOnCreateContextMenuListener(this);
+
+    }
+
+    @Override
+    public int getItemCount() {
+        return itemAccess.getCount();
+    }
+
+    @Override
+    public long getItemId(int position) {
+        NavDrawerData.DrawerItem drawerItem = itemAccess.getItem(position);
+        if (drawerItem != null && drawerItem.type == NavDrawerData.DrawerItem.Type.FEED) {
```

Comment:
I think this could cause issues with folders. How does the adapter that shows the side bar entries handle folder ids? Should be the same here.

Reply:
I'll see if there is issue.